dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -475,14 +510,25 @@ static void AddResponseMessage(ChatCompletionsOptions chatOptions, ChatHistory c
             }
 
             // Disable auto invocation if we've exceeded the allowed limit.
-            if (iteration >= chatExecutionSettings.ToolCallBehavior!.MaximumAutoInvokeAttempts)
+            if (requestIteration >= chatExecutionSettings.ToolCallBehavior!.MaximumAutoInvokeAttempts)
             {
                 autoInvoke = false;
                 if (this.Logger.IsEnabled(LogLevel.Debug))
                 {
                     this.Logger.LogDebug("Maximum auto-invoke ({MaximumAutoInvoke}) reached.", chatExecutionSettings.ToolCallBehavior!.MaximumAutoInvokeAttempts);
                 }
             }
+
+            // If filter requested to stop request iteration, returning latest result from LLM.
+            if (invocationContext is not null && invocationContext.Action.HasFlag(FunctionCallAction.StopRequestIteration))
```

Comment:
For a readability perspective I think it would be easier to understand if the flag indicated what we wanted to happen rather than what we want to stop e.g. `if (...HasFlag(ReturnToolCallResult))` is easier to understand than `if (...HasFlag(StopRequestIteration))`

Reply:
I simplified the implementation and removed Action enumeration completely in favour of simple boolean `Cancel` property.